File: example-android-views/app/src/main/java/co/verifyspeed/example/android/views/ui/common/VerificationDialog.java
package co.verifyspeed.example.android.views.ui.common;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.NonNull;

import co.verifyspeed.example.android.views.R;

public class VerificationDialog extends Dialog {
    private final String title;
    private final boolean isLoading;
    private final String phoneNumber;
    private final Runnable onDismiss;

    public VerificationDialog(
            @NonNull Context context,
            String title,
            boolean isLoading,
            String phoneNumber,
            Runnable onDismiss
    ) {
        super(context);
        this.title = title;
        this.isLoading = isLoading;
        this.phoneNumber = phoneNumber;
        this.onDismiss = onDismiss;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_verification);

        // Prevent dialog from being dismissed when touching outside
        setCanceledOnTouchOutside(false);
        setCancelable(false);

        // Initialize views
        TextView dialogTitle = findViewById(R.id.dialogTitle);
        ProgressBar progressBar = findViewById(R.id.progressBar);
        TextView phoneNumberText = findViewById(R.id.phoneNumberText);
        Button okButton = findViewById(R.id.okButton);

        // Set up the dialog
        dialogTitle.setText(title);
        
        // Handle visibility states
        if (isLoading) {
            progressBar.setVisibility(View.VISIBLE);
            phoneNumberText.setVisibility(View.GONE);
        } else if (phoneNumber != null) {
            progressBar.setVisibility(View.GONE);
            phoneNumberText.setVisibility(View.VISIBLE);
            
            // Format and display the phone number
            String formattedNumber = phoneNumber.trim();
            if (!formattedNumber.startsWith("+")) {
                formattedNumber = "+" + formattedNumber;
            }
            phoneNumberText.setText(String.format("Your verified phone number is:\n%s", formattedNumber));
        }

        okButton.setOnClickListener(v -> {
            dismiss();
            if (onDismiss != null) {
                onDismiss.run();
            }
        });
    }
} 

File: example/app/src/main/java/co/verifyspeed/example/ui/common/VerificationDialog.kt
package co.verifyspeed.example.ui.common

import androidx.compose.foundation.layout.size
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

@Composable
fun VerificationDialog(
    showDialog: Boolean,
    isLoading: Boolean,
    phoneNumber: String?,
    title: String,
    onDismiss: () -> Unit,
    onConfirm: () -> Unit = onDismiss
) {
    if (showDialog) {
        AlertDialog(
            onDismissRequest = onDismiss,
            title = { Text(title) },
            text = {
                if (isLoading) {
                    CircularProgressIndicator(modifier = Modifier.size(24.dp))
                } else {
                    Text("Your verified phone number is: $phoneNumber")
                }
            },
            confirmButton = {
                TextButton(onClick = onConfirm) { Text("OK") }
            }
        )
    }
} 